src/test/lib/ApiTest.cpp

```diff
@@ -1155,16 +1155,37 @@ DummyStreamCallback(
 struct CloseFromCallbackContext {
     uint16_t CloseCount;
     volatile uint16_t CurrentCount;
+    uint8_t RawBuffer[100];
+    QUIC_BUFFER BufferToSend { sizeof(RawBuffer), RawBuffer };
 
-    static QUIC_STATUS StreamCallback(_In_ MsQuicStream*, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT*) {
+    static QUIC_STATUS StreamCallback(_In_ MsQuicStream* Stream, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT* Event) {
+
+        if (Event->Type == QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE) {
+            Stream->Close(); // or abort
+        }
```

Comment:
You don't need this for auto cleanup/delete streams. It's done for you.

Reply:
This is really tricky and was magically working.

`CleanUpAutoDelete` does not only close the handle, but also `delete` the instance.
We cannot use `CleanUpAutoDelete` for client `MsQuicStream` created in main loop, because the instance is allocated on stack area.
However, the client stream get `QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE` by sending data with `QUIC_SEND_FLAG_FIN`, but it doesn't call `StreamClose()` by removing line:1164. Then `~MsQuicStream` is called at the end of loop which call `StreamClose()`. This get aborted because Connection is already closed.